lib/shopify-cli/commands/connect.rb

```diff
@@ -6,27 +6,20 @@ class Connect < ShopifyCli::Command
       def call(*)
         project_type = ask_project_type unless Project.has_current?
 
-        if Project.has_current? && Project.current && Project.current.env
+        if Project.has_current? && Project.current_project_type && Project.current.env
           @ctx.puts @ctx.message('core.connect.already_connected_warning')
           prod_warning = @ctx.message('core.connect.production_warning')
           @ctx.puts prod_warning if [:rails, :node].include?(Project.current_project_type)
         end
 
-        env_data = begin
-          Resources::EnvFile.parse_external_env
-                   rescue Errno::ENOENT
-                     {}
-        end
-
-        org = fetch_org
-        id = org['id']
-        app = get_app(id, org['apps'])
-        shop = get_shop(org['stores'], id)
-
-        write_env(app, shop, env_data[:scopes], env_data[:extra])
-        write_cli_yml(project_type, id) unless Project.has_current?
+        org = ShopifyCli::Tasks::EnsureEnv.new.call(@ctx, regenerate: true)
+        @api_key = Project.current.env['api_key']
+        write_cli_yml(project_type, org['id']) unless Project.has_current?
+        @ctx.puts(@ctx.message('core.connect.connected', get_app(org['apps']).first["title"]))
+      end
 
-        @ctx.puts(@ctx.message('core.connect.connected', app['title']))
+      def get_app(apps)
+        apps.select { |app| app["apiKey"] == @api_key }
```

Comment:
I think it probably makes more sense to have an `api_key` parameter instead of using an instance variable here